crates/biome_cli/tests/cases/linter_domains.rs

```diff
@@ -296,3 +296,62 @@ describe("foo", () => {
         result,
     ));
 }
+
+#[test]
+fn does_enable_test_rules_when_recommended_rules_are_disabled_but_domain_is_enabled() {
+    let mut console = BufferConsole::default();
+    let mut fs = MemoryFileSystem::default();
+    let config = Utf8Path::new("biome.json");
+    fs.insert(
+        config.into(),
+        r#"{
+    "linter": {
+        "rules": {
+            "recommended": false
+        },
+        "domains": {
+            "test": "all"
+        }
+    }
+}
+"#
+        .as_bytes(),
+    );
+    let test1 = Utf8Path::new("test1.js");
+    fs.insert(
+        test1.into(),
+        r#"
+        debugger;
```

Comment:
As you can notice, the rule `noDebugger` isn't triggered, even though it's recommended.